### PR TITLE
Iprep/v1 bug 6834 fix 0 reputation value handling

### PR DESCRIPTION
**Pull request by victorjulien**
SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1799

https://redmine.openinfosecfoundation.org/issues/6834

### CODE REVIEW

File: src/detect-iprep.c [hunk 1]
```diff
@@ -75,65 +75,66 @@ void DetectIPRepRegister (void)
...
      * rule reload */
     if (r != NULL && r->version >= version) {
         return r->rep[cat];
     }
-    return 0;
+    return -1;
 }
 
-static uint8_t GetHostRepSrc(Packet *p, uint8_t cat, uint32_t version)
+/** \returns: -2 no host, -1 no rep entry, 0-127 rep values */
```
Comment (jasonish) [inline src/detect-iprep.c:88]: 1) Could -1 be used in the rule for does not exist?
2) Documentation needs an update to say 0 is allowed, currently its 1-127.

Comment (victorjulien) [reply src/detect-iprep.c:88]: Doc updated.

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 20350

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .memcap.pressure | 57 | 55 | 96.49% |
| .memcap.pressure_max | 62 | 64 | 103.23% | 

Pipeline 20353

**Comment by catenacyber** [comment]
Draft because of push 

I guess a new PR rebased to master should get green CI

**Comment by victorjulien** [comment]
replaced by #11000

**Pull request closed without merging**